### PR TITLE
Minor: Update quarterly_roadmap.md

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

N/A

## Rationale for this change

In general I think it would be good for us to communicate more broadly what we are working on, for at least two reasons:
1. So people who may be interested / impacted have a chance to weigh in on designs
2. To help find people to help with projects that may need additional assistance

I would love additional ideas about how to do this -- maybe weekly summaries of "this week in DataFusion" 🤔  Any takers for doing so?

## What changes are included in this PR?

Update quarterly roadmap. See rendered version here: https://github.com/apache/arrow-datafusion/blob/alamb-patch-1/docs/source/contributor-guide/quarterly_roadmap.md

### CODE REVIEW

File: docs/source/contributor-guide/quarterly_roadmap.md [hunk 1]
```diff
@@ -23,9 +23,22 @@ A quarterly roadmap will be published to give the DataFusion community visibilit
 
 ## 2023 Q4
 
+### Underway
+
+- Improved sort based optimizations (TODO FIND TICKET)
 - Improve data output (`COPY`, `INSERT` and DataFrame) output capability [#6569](https://github.com/apache/arrow-datafusion/issues/6569)
 - Implementation of `ARRAY` types and related functions [#6980](https://github.com/apache/arrow-datafusion/issues/6980)
 - Write an industrial paper about DataFusion for SIGMOD [#6782](https://github.com/apache/arrow-datafusion/issues/6782)
+- Faster Merging with Parallel Cascaded Merge: [#7181](https://github.com/apache/arrow-datafusion/issues/7181)
```
Comment (alamb) [inline docs/source/contributor-guide/quarterly_roadmap.md:32]: FYI @wiedld

File: docs/source/contributor-guide/quarterly_roadmap.md [hunk 2]
```diff
@@ -23,9 +23,22 @@ A quarterly roadmap will be published to give the DataFusion community visibilit
...
+
+### Planning (needs additional help, guidance)
+
+The following work items are still in the planning / proposal stage.
+
+- Improved planning speed, especially for schemas with large numbers of columns [#5637](https://github.com/apache/arrow-datafusion/issues/5637) / [#7698](https://github.com/apache/arrow-datafusion/issues/7698)
+- Make it easier to customize DataFusion, with split out function packages, etc. - [#7977](https://github.com/apache/arrow-datafusion/issues/7977)
+- User Defined Types [#7923](https://github.com/apache/arrow-datafusion/issues/7923)
```
Comment (alamb) [inline docs/source/contributor-guide/quarterly_roadmap.md:40]: fyi @yukkit

File: docs/source/contributor-guide/quarterly_roadmap.md [hunk 3]
```diff
@@ -23,9 +23,22 @@ A quarterly roadmap will be published to give the DataFusion community visibilit
...
+- ORC Support [#4707](https://github.com/apache/arrow-datafusion/issues/4707)
```
Comment (alamb) [inline docs/source/contributor-guide/quarterly_roadmap.md:41]: FYI @Jefffrey  / @waynexia

File: docs/source/contributor-guide/quarterly_roadmap.md [hunk 4]
```diff
@@ -23,9 +23,22 @@ A quarterly roadmap will be published to give the DataFusion community visibilit
...
 - Implementation of `ARRAY` types and related functions [#6980](https://github.com/apache/arrow-datafusion/issues/6980)
 - Write an industrial paper about DataFusion for SIGMOD [#6782](https://github.com/apache/arrow-datafusion/issues/6782)
+- Faster Merging with Parallel Cascaded Merge: [#7181](https://github.com/apache/arrow-datafusion/issues/7181)
+
+### Planning (needs additional help, guidance)
+
+The following work items are still in the planning / proposal stage.
+
+- Improved planning speed, especially for schemas with large numbers of columns [#5637](https://github.com/apache/arrow-datafusion/issues/5637) / [#7698](https://github.com/apache/arrow-datafusion/issues/7698)
+- Make it easier to customize DataFusion, with split out function packages, etc. - [#7977](https://github.com/apache/arrow-datafusion/issues/7977)
```
Comment (alamb) [inline docs/source/contributor-guide/quarterly_roadmap.md:39]: fyi @2010YOUY01

**Review by alamb — COMMENTED** [review]
There are several potential non trivial changes underway that I think would benefit significantly from additional help - both design feedback, as well as code review, and assistance communicating, etc.

I would really love to help them all but I am running out of time, and I want to be as clear as possible that I am unlikely to be able to help with all of this.

My ask for other maintainers to offer to help guide contributors through some of the following efforts, ideally focused on areas that are aligned with their interests

File: docs/source/contributor-guide/quarterly_roadmap.md [hunk 5]
```diff
@@ -23,9 +23,22 @@ A quarterly roadmap will be published to give the DataFusion community visibilit
 
 ## 2023 Q4
 
+### Underway
+
+- Improved sort based optimizations (TODO FIND TICKET)
```
[the remark this answers is not in the document]
Comment (mustafasrepo) [reply docs/source/contributor-guide/quarterly_roadmap.md:28]: I have opened an issue for this feature. I will put a link from relevant PRs to this issue.
Issue is [#8064](https://github.com/apache/arrow-datafusion/issues/8064)

[the remark this answers is not in the document]
Comment (alamb) [reply docs/source/contributor-guide/quarterly_roadmap.md:28]:
```suggestion
- Improved sort based optimizations [#8064](https://github.com/apache/arrow-datafusion/issues/8064)
```

**Comment by alamb** [comment]
This seems not to have garnered much attention, so let's just shut it down

**Pull request closed without merging**